src/python/pants/backend/native/config/environment.py

```diff
@@ -132,10 +141,7 @@ class CCompiler(datatype([
 
   @property
   def as_invocation_environment_dict(self):
-    ret = super(CompilerMixin, self).as_invocation_environment_dict.copy()
-
-    if self.include_dirs:
-      ret['C_INCLUDE_PATH'] = create_path_env_var(self.include_dirs)
+    ret = super(CCompiler, self).as_invocation_environment_dict.copy()
 
     ret['CC'] = self.exe_filename
```

Comment:
With the correct `exe_filename`? There is no testing that the setup.py compilation does anything other than succeed, currently, but if you make the build fail (e.g. by inserting a syntax error into a C/C++ source), pex will print the stdout and stderr to the terminal stderr, and you can see that modifying `exe_filename`s (or e.g. selecting `GCCCToolchain` instead of the llvm one in `python_native_code.py`) will change the compiler used for the generated command line to build setup.py native sources.

At this point, I'm not sure what we should be testing wrt setup.py compilation other than success. If you have some itches you want to scratch, an issue just listing what tests we should add, or a PR doing that, would be great so I or someone else can address it in full. This wouldn't need to more than a few sentences.

Reply:
And that may have been the case (wrt not respecting `CC`), but I don't remember the situation, and the issue may have been e.g. that we were adding the wrong entries to the `PATH`.